src/Capability/Constraints.hs

```diff
@@ -16,12 +16,12 @@ module Capability.Constraints
   ) where
 
 import Data.Constraint (Dict(..))
-import Data.Kind (Constraint)
+import Data.Kind (Constraint, Type)
 
--- | A 'Capability' takes a type constructor @* -> *@ (e.g., a monad) and
+-- | A 'Capability' takes a type constructor @Type -> Type@ (e.g., a monad) and
 -- returns a 'Constraint'. Examples of capabilities includ: @HasReader "foo"
 -- Int@, @MonadIO@, …
-type Capability = (* -> *) -> Constraint
+type Capability = (Type -> Type) -> Constraint
```

Comment:
It probably would have been less noisy to enable `-XStarIsKind`. But at least, now, we are more future-proof.